### PR TITLE
[15.0][MIG] event_mail: Migration to version 15.0

### PR DESCRIPTION
**Pull request by ernestotejeda**
Cc @Tecnativa TT42036

### CODE REVIEW

File: event_mail/models/event_mail.py [hunk 1]
```diff
@@ -28,14 +38,16 @@ def _default_scheduler_template_ids(self):
...
+                % self.env.ref("event.event_subscription").id,
             },
             {
                 "notification_type": "mail",
                 "interval_nbr": 10,
                 "interval_unit": "days",
                 "interval_type": "before_event",
-                "template_id": self.env.ref("event.event_reminder").id,
+                "template_ref": "mail.template, %i"
+                % self.env.ref("event.event_reminder").id,
```
Comment (chienandalu) [inline event_mail/models/event_mail.py]:
```suggestion
                "template_ref": f"mail.template, {self.env.ref("event.event_reminder").id}",

File: event_mail/tests/test_event_mail.py [hunk 2]
```diff
@@ -86,7 +87,7 @@ def test_event_template_no_config(self):
         event = self.env["event.event"].create(vals)
         self.assertEqual(
             len(event.event_mail_ids),
-            0,
+            len(event._default_event_mail_ids()),
```
Comment (chienandalu) [inline event_mail/tests/test_event_mail.py]: :thinking: tests assertions should be deterministic

**Comment by ernestotejeda** [comment]
@chienandalu could you review again?

File: event_mail/tests/test_event_mail.py [hunk 3]
```diff
@@ -86,7 +87,7 @@ def test_event_template_no_config(self):
         event = self.env["event.event"].create(vals)
         self.assertEqual(
             len(event.event_mail_ids),
-            0,
+            3,
             "Event Mail: mails scheduler no created for this event",
```
Comment (chienandalu) [inline event_mail/tests/test_event_mail.py]: But then, was the test wrong before? :thinking:

Comment (ernestotejeda) [reply event_mail/tests/test_event_mail.py]: Here the default value of the `event_mail_ids` field in events is being testing when no template is defined in the res.config.setting. In v13, the event was created with an empty `event_mail_ids`, but in v15, it takes what the `_default_event_mail_type_ids` method of `event.type` returns  https://github.com/OCA/OCB/blob/15.0/addons/event/models/event_event.py#L27 and creates 3 records for `event_mail_ids`.

**Comment by chienandalu** [comment]
please review @stefan-tecnativa

File: event_mail/tests/test_event_mail.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        vals = {
+            "name": "Event test",
+            "date_begin": "2017-05-01",
+            "date_end": "2017-06-01",
+            "auto_confirm": False,
+        }
+        event = self.env["event.event"].create(vals)
+        self.assertEqual(
+            len(event.event_mail_ids),
+            3,
```
Comment (stefan-tecnativa) [inline event_mail/tests/test_event_mail.py:90]:
```suggestion
            len(event.event_type_id._default_event_mail_type_ids()),
```

As the event hasn't any type defined by default, we could compare the mails scheduled for the event with those that are scheduled by default when the type is not defined for the event. @chienandalu what do you think about this?

File: event_mail/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
+To use this module, you need to:
+
+#. Go to *Events > Configuration > Mail Templates* and create all templates
+   that you need.
+#. Go to *Events > Settings* and select the default template in
+   "Template Mail Scheduler" field.
+#. Go to *Events > Events* and create one. All mails schedulers has been
+   created. Also you can select other template in "Mail Template Scheduler"
+   field on "Email Schedule" tab.
```
Comment (stefan-tecnativa) [inline event_mail/readme/USAGE.rst]:
```suggestion
   field on "Communication" tab.
```

**Review by stefan-tecnativa — COMMENTED** [review]
Functionally tested, LGTM. Just check the comments.

**Comment by ernestotejeda** [comment]
@stefan-tecnativa changes done

File: event_mail/tests/test_event_mail.py (hunk 4 above)
Comment (chienandalu) [reply event_mail/tests/test_event_mail.py:90]: I'm ok with the suggestion :+1:

**Review by chienandalu — APPROVED** [review]
Thanks everyone :)

**Review by ivantodorovich — APPROVED** [review]
LG

**Pull request merged**